packages/react-dom/src/__tests__/ReactDOMHooks-test.js

```diff
@@ -105,6 +109,145 @@ describe('ReactDOMHooks', () => {
+        </div>
+      );
+    }
+
+    const root = ReactDOM.createRoot(container);
+    await act(async () => {
+      root.render(<App />);
+    });
+
+    expect(container.innerHTML).toMatchInlineSnapshot(
+      `"<div><div aria-labelledby=\\"c_1\\"></div></div>"`,
+    );
+
+    await act(async () => {
+      _setShowId(true);
+    });
+
+    expect(container.innerHTML).toMatchInlineSnapshot(
+      `"<div><div aria-labelledby=\\"c_1\\"></div><div id=\\"c_1\\"></div></div>"`,
+    );
+  });
+
+  it('useUniqueID: IDs match when, after hydration, a new component that uses the ID is rendered', async () => {
+    const {act} = TestUtils;
+    const {useUniqueID, useState} = React;
+    let _setShowDiv;
+    function App() {
+      const id = useUniqueID();
+      const [showDiv, setShowDiv] = useState(false);
+      _setShowDiv = setShowDiv;
+
+      return (
+        <div>
+          <div id={id}>Child One</div>
+          {showDiv && <div id={id}>Child Two</div>}
+        </div>
+      );
+    }
+
+    container.innerHTML = ReactDOMServer.renderToString(<App />);
+    const root = ReactDOM.createRoot(container, {hydrate: true});
+    await act(async () => {
+      root.render(<App />);
+    });
+    expect(container.innerHTML).toMatchInlineSnapshot(
+      `"<div data-reactroot=\\"\\"><div id=\\"s_0\\">Child One</div></div>"`,
+    );
+    await act(async () => {
+      _setShowDiv(true);
+    });
+    expect(container.innerHTML).toMatchInlineSnapshot(
+      `"<div data-reactroot=\\"\\"><div id=\\"c_0\\">Child One</div><div id=\\"c_0\\">Child Two</div></div>"`,
+    );
+  });
+
+  it('useUniqueID: IDs match when part of the DOM tree is server rendered and part is client rendered', async () => {
+    const {useUniqueID} = React;
+    let suspend = false;
+    let resolve;
+    let promise = new Promise(resolvePromise => (resolve = resolvePromise));
+
+    function Child({text}) {
+      if (suspend) {
+        throw promise;
+      } else {
+        return text;
+      }
+    }
+
+    function App() {
+      const id = useUniqueID();
+      return (
+        <div>
+          <div id={id}>Child One</div>
+          <React.Suspense fallback={'Fallback'}>
+            <div id={id}>
+              <Child text="Child Two" />
+            </div>
+          </React.Suspense>
+        </div>
+      );
+    }
+
+    container.innerHTML =
+      '<div data-reactroot=""><div id="s_0">Child One</div><!--$!--><div>Fallback</div><!--/$--></div>';
+
+    suspend = true;
+    const root = ReactDOM.createRoot(container, {hydrate: true});
+    root.render(<App />);
+    Scheduler.unstable_flushAll();
+    jest.runAllTimers();
+
+    expect(container.innerHTML).toMatchInlineSnapshot(
+      `"<div data-reactroot=\\"\\"><div id=\\"s_0\\">Child One</div>Fallback</div>"`,
+    );
+
+    suspend = false;
+    resolve();
+    await promise;
+
+    Scheduler.unstable_flushAll();
+    jest.runAllTimers();
+
+    expect(container.innerHTML).toMatchInlineSnapshot(
```

Comment:
The output is correct, but let's also test that there was only a single commit. You can do that with `useEffect`:

```js
const id = useUniqueID();
useEffect(() => {
  Scheduler.unstable_yieldValue('Did commit');
});
```

I expect this test will fail; let's write the test first and then I can work with you on how to fix it.